pkg/prometheus/server/statefulset.go

```diff
@@ -326,13 +326,22 @@ func makeStatefulSetSpec(
 		return nil, err
 	}
 
+	envVars := []v1.EnvVar{}
+	// For higher Prometheus version its set with runtime field in configuration
+	if p.Spec.Runtime != nil && cg.Version().LTE(semver.MustParse("2.52.1")) {
+		if p.Spec.Runtime.GoGC != nil {
+			envVars = append(envVars, v1.EnvVar{Name: "GOGC", Value: fmt.Sprintf("%d", *p.Spec.Runtime.GoGC)})
```

Comment:
need to check GoGC for nil too

Reply:
did you mean we need to set default value (current prometheus default is 75).?

Reply:
IIUC we don't want to define default value API level for applying default for >=v2.53.0?